EIPS/eip-5744.md

```diff
@@ -0,0 +1,97 @@
+---
+eip: 5744
+title: Latent Fungible Token
+description: An interface for tokens that become fungible after a period of time.
+author: Cozy Finance (@cozyfinance), Tony Sheng (@tonysheng), Matt Solomon (@mds1), David Laprade (@davidlaprade), Payom Dousti (@payomdousti), Chad Fleming (@chad-js), Franz Chen (@Dendrimer)
+discussions-to: https://ethereum-magicians.org/t/eip-5744-latent-fungible-token/11111
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-09-29
+requires: 20, 2612
+---
+
+## Abstract
+
+The following standard is an extension of [EIP-20](./eip-20.md) that enables tokens to become fungible after some initial non-fungible period.
+Once minted, tokens are non-fungible until they reach maturity.
+At maturity, they become fungible and can be transferred, traded, and used in any way that a standard EIP-20 token can be used.
+
+## Motivation
+
+Example use cases include:
+
+- Receipt tokens that do not become active until a certain date or condition is met. For example, this can be used to enforce minimum deposit durations in lending protocols.
+- Vesting tokens that cannot be transferred or used until the vesting period has elapsed.
+
+## Specification
+
+All [EIP-5744](./eip-5744.md) tokens MUST implement EIP-20 to represent the token.
+The `balanceOf` and `totalSupply` return quantities for all tokens, not just the matured, fungible tokens.
+A new method called `balanceOfMatured` MUST be added to the ABI.
+This method returns the balance of matured tokens for a given address:
+
+```solidity
+function balanceOfMatured(address user) external view returns (uint256);
+```
+
+An additional method called `getMints` MUST be added, which returns an array of all mint metadata for a given address:
+
+```solidity
+struct MintMetadata {
+  // Amount of tokens minted.
+  uint256 amount;
+  // Timestamp of the mint, in seconds.
+  uint256 time;
+  // Delay in seconds until these tokens mature and become fungible. When the
+  // delay is not known (e.g. if it's dependent on other factors aside from
+  // simply elapsed time), this value must be `type(uint256).max`.
+  uint256 delay;
+}
+
+function getMints(address user) external view returns (MintMetadata[] memory);
+```
+
+Note that the implementation does not require that each of the above metadata parameters are stored as a `uint256`, just that they are returned as `uint256`.
+
+An additional method called `mints` MAY be added.
+This method returns the metadata for a mint based on it's ID:
+
+```solidity
+function mints(address user, uint256 id) external view returns (MintMetadata memory);
+```
+
+The ID is not prescriptive&mdash;it may be an index in an array, or may be generated by other means.
```

Comment:
Please use the mdash unicode character.

Reply:
Done in b9b4f60f. Also committed your other suggestions from the GitHub UI which automatically marked those comments resolved